SwiftCheck/Gen.swift

```diff
@@ -19,9 +23,99 @@ public struct Gen<A> {
 		return unGen(r)(30)
 	}
 
-	/// These must be in here or we get linker errors; rdar://21172325
+	/// Constructs a Generator that selects a random value from the given list and produces only 
+	/// that value.
+	///
+	/// The input array is required to be non-empty.
+	public static func fromElementsOf(xs : [A]) -> Gen<A> {
+		assert(xs.count != 0, "Gen.fromElementsOf used with empty list")
+
+		return choose((0, xs.count - 1)).fmap { i in
+			return xs[i]
+		}
+	}
+
+	/// Constructs a Generator that uses a given array to produce smaller arrays composed of its 
+	/// initial segments.  The size of each initial segment increases with the receiver's size 
+	/// parameter.
+	///
+	/// The input array is required to be non-empty.
+	public static func fromInitialSegmentsOf(xs : [A]) -> Gen<A> {
```

Comment:
These `from*` functions feel out of place with the other constructors, but I couldn't quite think of a good verb.  Ideas?